### PR TITLE
Handle words missing from word2vec model

### LINKED ISSUE #9: Handle "missing word" cases
Unlike in the benchmarking scenarios, spec2vec will probably often be used like this:
- A small to moderate number of spectra is converted into documents.
- A word2vec model that was trained on a much larger dataset will be imported.
- Spectrum vectors (and based on that: similarities) will be calculated for the new query spectrums.
 
Even if the model was trained on a very large corpus/dataset, it can still occur that a few "words" of the new query spectrums are not yet in the model. Two strategies should be implemented to deal with this:
1. Short retraining of the model on the new data. Or:
2. Ignore missing words. (safer: ignore if word-intensities are below a set threshold)

### PR DESCRIPTION
**Pull request by florian-huber**
--> Best to merge master after first merging PR #6 to make docstrings consistent.

- major changes to ``calc_vector``: now has ``allowed_missing_percentage`` parameter. If that is >0 the function can also calculate document/spectrum vectors for cases when (a few) words are missing from the pretrained model.
- The function now also checks if the intensity are properly normalized. That lead to changes in matchms so I had to adapt the integration test accordingly (first apply normalization, then ``add_losses``).
- Add test for ``calc_vector`` (it feels a bit in-between integration and unit test though)

### CODE REVIEW

File: spec2vec/calc_vector.py [hunk 1]
```diff
@@ -1,10 +1,58 @@
...
+
+    word_vectors = model.wv[words_in_model]
+    weights_raised = numpy.power(weights_in_model, intensity_weighting_power)
+
+    if _sufficient_model_coverage():
+        weights_raised_tiled = numpy.tile(weights_raised, (1, model.wv.vector_size))
+        vector = numpy.sum(word_vectors * weights_raised_tiled, 0)
+        return vector
+
+    return None
```
Comment (sverhoeven) [inline spec2vec/calc_vector.py]: Why return None if coverage is not sufficient? Would a raised exception with an assert not be more useful?
The user has to check the results for Nones or are use try except a exception is more Pythonic.

File: spec2vec/Spec2Vec.py [hunk 2]
```diff
@@ -1,18 +1,58 @@
...
+        Parameters
+        ----------
+        model : gensim word2vec model
+            Expecgted input is a gensim word2vec model that has been trained on
+            the desired set of spectrum documents.
+        intensity_weighting_power : float, optional
+            Spectrum vectors are a weighted sum of the word vectors. The given
+            word intensities will be raised to the given power.
+            The default is 0, which means that no weighing will be done.
+        allowed_missing_fraction:
```
Comment (sverhoeven) [inline spec2vec/Spec2Vec.py]: If you name a var fraction I expect to use 0..1 a fraction as value. Could you either use a real fraction everywhere or change var name to `allowed_missing_percentage`?

**Review by sverhoeven — CHANGES_REQUESTED** [review]

File: spec2vec/Spec2Vec.py (hunk 2 above)
Comment (florian-huber) [reply spec2vec/Spec2Vec.py]: Changed it to ``allowed_missing_percentage``.

File: spec2vec/calc_vector.py (hunk 1 above)
Comment (florian-huber) [reply spec2vec/calc_vector.py]: I added an assert and removed the ``return None`` part.

**Review by sverhoeven — APPROVED** [review]
Nice you remembered to update CHANGELOG.

Tests could be improved with tests for the assertions in calc_vector, but that can also be done in a PR to improve the branch coverage.

**Comment by florian-huber** [comment]
Does now contain test for the assertion in calc_vector.

**Comment by florian-huber** [comment]
Thanks for reviewing!

**Pull request merged**